### PR TITLE
SR-7039: Add test for -static-stdlib when using Dispatch

### PR DESCRIPTION
**Pull request by spevans**

### CODE REVIEW

**Comment by spevans** [comment]
Test requires https://github.com/apple/swift/pull/25085 and https://github.com/apple/swift-corelibs-libdispatch/pull/489

**Comment by spevans** [comment]
@swift-ci test

**Comment by spevans** [comment]
@weissi Could you test and merge this please I have commit access to this repo, thanks

**Comment by weissi** [comment]
@kevints is doing the release management for this release

**Comment by weissi** [comment]
@swift-ci test

File: test-static-stdlib/test-dispatch-static-stdlib.test [hunk 1]
```diff
@@ -0,0 +1,7 @@
+REQUIRES: platform=Linux
+RUN: rm -rf %t
+RUN: mkdir -p %t
+RUN: %{swiftc} -static-stdlib %S/dispatch_test.swift -lDispatchStubs -lbsd -o %t/dispatch_test
```
Comment (jrose-apple) [inline test-static-stdlib/test-dispatch-static-stdlib.test]: `-lDispatchStubs` being necessary seems very suspect to me. Shouldn't that be handled by Dispatch itself?

Comment (kevints) [reply test-static-stdlib/test-dispatch-static-stdlib.test]: > @kevints is doing the release management for this release

This is targeted at `master`, so this should be reviewed by code owners instead.

Comment (kevints) [reply test-static-stdlib/test-dispatch-static-stdlib.test]: For that matter `-lbsd` seems suspect. Is the problem a missing modulemap for libbsd?

Comment (spevans) [reply test-static-stdlib/test-dispatch-static-stdlib.test]: `libdispatch.a` is an archive library so wont auto link `libswiftDispatch.a` and `libDispatchStubs.a` hence being explicitly listed. 

`-lbsd` was needed the last time I tested this to provide `strlcpy` and `getprogname` however it looks like they are now included as shims in the dispatch source tree so I have removed `-lbsd` from the test.

@kevints @jrose-apple Could someone please re-CI test  this PR, thanks

Comment (jrose-apple) [reply test-static-stdlib/test-dispatch-static-stdlib.test]: Yeah, @kevints reminded me about what we can and can't do with static libraries. I just wasn't thinking.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci test

**Comment by spevans** [comment]
@shahmishal Is it possible to run tests in this repo?

**Comment by shahmishal** [comment]
We don't support PR testing on this repository yet.

**Comment by spevans** [comment]
I tested this PR using another one https://github.com/apple/swift-corelibs-foundation/pull/2335 and the log is here: https://ci.swift.org/job/swift-corelibs-foundation-PR-Linux/3441/consoleFull

Looks like it passed ok so I think its ok to merge now.

**Pull request merged**